tests/testthat/test-methods.R

```diff
@@ -147,3 +148,9 @@ test_that("split.lint works as intended", {
   l <- lint(tmp, seq_linter())
   expect_true(all(vapply(split(l), inherits, logical(1L), "lints")))
 })
+
+test_that("within.list is dispatched", {
+  l <- lint(text = "a=1\nb=2", linters = infix_spaces_linter())
+  expect_silent(l <- lapply(l, within, line_number <- line_number + 1L))
+  expect_identical(vapply(l, `[[`, integer(1L), "line_number"), 2:3)
```

Comment:
I know, this is a bit annoying.

```suggestion
  expect_identical(vapply(l, `[[`, integer(1L), "line_number"), 2L:3L)
```